src/multiCursor.ts

```diff
@@ -17,3 +17,26 @@ const addCursorFactory =
 
 export const addCursorUp = addCursorFactory('up');
 export const addCursorDown = addCursorFactory('down');
+
+export const addCursorAtEachSelectionLine: Command = (view) => {
+  const cursors = view.state.selection.ranges.reduce((acc, r) => {
+    if (r.empty) {
+      return acc;
+    }
+
+    for (let pos = r.from; pos <= r.to; ) {
+      const line = view.state.doc.lineAt(pos);
+      acc = [...acc, EditorSelection.cursor(line.to > r.to ? r.to : line.to)];
```

Comment:
probably worth using addRange here too: https://codemirror.net/6/docs/ref/#state.EditorSelection.addRange

also you can use `Math.max` here?

Reply:
So like start my reduce with an empty `EditorSelection` and use `addRange`? Sure :shrug:

Reply:
`Math.min` is what I'm looking for

Reply:
Doesn't look like there's a way to create an empty selection and then add ranges, I'll keep as is.

Sent an update to use Math.min